mPowerSDK/Dashboard/APHLineGraphView.m

```diff
@@ -70,4 +103,138 @@ - (void)drawXAxis
     }
 }
 
+- (void)drawGraphForPlotIndex:(NSInteger)plotIndex {
+	[super drawGraphForPlotIndex:plotIndex];
+
+	if (self.shouldDrawLastPoint) {
+		[self drawLastPointDot:plotIndex];
+	}
+}
+
+- (void)drawLastPointDot:(NSInteger)plotIndex {
+	__block NSInteger i = -1;
+	[self.dataPoints enumerateObjectsUsingBlock:^(NSNumber *dataPoint, NSUInteger idx, BOOL *stop) {
+		if (dataPoint.floatValue != NSNotFound) {
+			i = idx;
+		}
+	}];
+	
+	if (i < 0) return;
```

Comment:
Refactor finding `i` into a method so that the early exit is more obvious. Also, add curly braces, please.